docs/meta.md

```diff
@@ -0,0 +1,116 @@
+| `engine.args`                    | String   | Yes       | Arguments to start the blocklet script                    |                                                                                                        | Draft  |
+| `engine.platform`                | String   | No        | Which platform can use this script                        |                                                                                                        | Draft  |
+|                                |          |           |                                                           |                                                                                                        | Draft  |
+| `dist`                          | Object   | No        | Blocklet distribution info                                |                                                                                                        | Draft  |
+| `dist.tarball`                   | String   | Yes       | URL to download the blocklet tarball                      | Can use underlying npm registry, but can support any valid url                                         | Draft  |
+| `dist.integrity`                 | String   | Yes       | Used to verify the download                               | Refer to [SRI](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)            | Draft  |
+| `dist.file_count`                | Number   | Yes       | Files included in the bundle                              |                                                                                                        | Draft  |
+| `dist.unpacked_size`             | Number   | Yes       | Size on disk of the package when unpacked.                |                                                                                                        | Draft  |
+| `dist.registry_pk`               | String   | No        | Public key of the registry                                |                                                                                                        | Draft  |
+| `dist.registry_did`              | String   | No        | DID of the registry                                       |                                                                                                        | Draft  |
+| `dist.registry_sig`              | String   | No        | Signature for the blocklet by the registry                |                                                                                                        | Draft  |
+
+> For integrity generating and verifying, we can leverage the [ssri](https://www.npmjs.com/package/ssri) package.
+
+## Blocklet DID Generation
+
+1. Take `name` field from blocklet meta
+2. Create buffer from name: `Buffer.from(name, 'utf8')`
+3. Use buffer as the public-key to generate the did: `fromPublicKey(buffer, { role: types.RoleType.Any })
+
+## Blocklet Bundle Format
```

Comment:
Let's move the bundle format to a separate document.  Many developers may not care about how bundle is made, but will need to know blocklet.yml format